js/common/util/state_card_type.js

```diff
@@ -0,0 +1,31 @@
+import canToggleState from './can_toggle_state.js';
+import computeStateDomain from './compute_state_domain.js';
+
+const DOMAINS_WITH_CARD = [
+  'climate',
+  'cover',
+  'configurator',
+  'input_select',
+  'input_number',
+  'input_text',
+  'media_player',
+  'scene',
+  'script',
+  'weblink',
+];
+
+export default function stateCardType(hass, stateObj) {
+  if (stateObj.state === 'unavailable') {
+    return 'display';
+  }
+
+  const domain = computeStateDomain(stateObj);
+
+  if (DOMAINS_WITH_CARD.includes(domain)) {
```

Comment:
`includes` is not supported in IE https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/includes

Reply:
(and it also won't be transpiled)

Reply:
How do you feel about a polyfill in compatibility.js?

https://www.npmjs.com/package/array-includes

Reply:
Look at all the dependencies and code of that polyfill. That's too much. The polyfill in the MDN docs is better but honestly, what's wrong with using `indexOf`?

Reply:
Mostly just a matter of readability for me. Argument summed up here well: https://stackoverflow.com/a/35370411

I'd also prefer to ship a polyfill just because I know I've personally been approaching this with the mindset of "modern JS features are safe to use", and would prefer to not have to keep exceptions in mind while writing and reviewing code.

Reply:
Alright. Let's use the MDN polyfill and add it to compatability.js